engine/admin-api/domain/repository/process_registry.go

```diff
@@ -9,6 +9,7 @@ import (
 )
 
 type ProcessRegistryRepo interface {
-	Create(productID string, newProcessRegistry *entity.ProcessRegistry) (*entity.ProcessRegistry, error)
 	CreateIndexes(ctx context.Context, productID string) error
+	Create(productID string, newProcessRegistry *entity.ProcessRegistry) (*entity.ProcessRegistry, error)
+	ListByProductWithTypeFilter(ctx context.Context, productID, processType string) ([]*entity.ProcessRegistry, error)
```

Comment:
Both product and type are filters for the query, I'd rename the method to `ListByProductAndType` to make it more clear.

```suggestion
	ListByProductAndType(ctx context.Context, productID, processType string) ([]*entity.ProcessRegistry, error)
```